library/src/test_utils.rs

```diff
@@ -0,0 +1,45 @@
+/// Helper methods for tests.
+use std::fs;
+
+use crate::{
+    cache::{PatchInfo, UpdaterState},
+    config::with_config,
+};
+
+/// Writes a fake patch to the patches directory and sets it as the next boot patch.
+pub fn install_fake_patch(patch_number: usize) -> anyhow::Result<()> {
+    with_config(|config| {
+        let download_dir = std::path::PathBuf::from(&config.download_dir);
+        let artifact_path = download_dir.join(patch_number.to_string());
+        fs::create_dir_all(&download_dir)?;
+        fs::write(&artifact_path, "hello")?;
+
+        let mut state = UpdaterState::load_or_new_on_error(
+            &config.storage_dir,
+            &config.release_version,
+            config.patch_public_key.as_deref(),
+        );
+        state.install_patch(
+            &PatchInfo {
+                path: artifact_path,
+                number: patch_number,
+            },
+            "hash",
+            None,
+        )?;
+        state.save()
+    })
+}
+
+/// Creates a fake APK at `apk_path` and writes `libapp_contents` to its relative `libapp.so` path.
+pub fn write_fake_apk(apk_path: &str, libapp_contents: &[u8]) {
+    use std::io::Write;
+    let mut zip = zip::ZipWriter::new(std::fs::File::create(apk_path).unwrap());
+    let options = zip::write::FileOptions::default()
+        .compression_method(zip::CompressionMethod::Stored)
+        .unix_permissions(0o755);
```

Comment:
is 0 correct?  Or do you want x?

Reply:
I'm also not sure why the fake apk needs executable permissions?

Reply:
Me neither tbh—this function was just moved from a [much earlier commit](https://github.com/shorebirdtech/updater/commit/226df7d08dd7899e5cad6b597f2405708e8560d4). 

According to the `unix_permissions` docs, they seem to want octal:

```
    /// The format is represented with unix-style permissions.
    /// The default is `0o644`, which represents `rw-r--r--` for files,
    /// and `0o755`, which represents `rwxr-xr-x` for directories
```

It looks like we might've just used the directory default, possibly through copilot? The tests pass without specifying a permission, so I'm just going to remove this line.